discovery/gossiper.go

```diff
@@ -981,12 +986,16 @@ func (d *AuthenticatedGossiper) networkHandler() {
 				// the emitted announcements to our announce
 				// batch to be broadcast once the trickle timer
 				// ticks gain.
-				if emittedAnnouncements != nil {
+				if emittedAnnouncements != nil && shouldBroadcast {
```

Comment:
will this skip broadcasting both local and remote anns? shouldn't local ones be allowed through?

Reply:
Good catch, didn't think of this case. Fixed.